File: DotNetClient/Guts.Client.Shared/Models/AssignmentTestRun.cs
using System.Collections.Generic;

namespace Guts.Client.Shared.Models
{
    public class AssignmentTestRun
    {
        public Assignment Assignment { get; set; }

        public IEnumerable<TestResult> Results { get; set; }

        public IEnumerable<SolutionFile> SolutionFiles { get; set; }

        public string TestCodeHash { get; set; }
    }
}

File: JavaClient/guts-java-client/src/main/java/guts/client/models/AssignmentTestRun.java
package guts.client.models;

import java.util.List;

public class AssignmentTestRun {

    private Assignment assignment;
    private List<TestResult> results;
    private String sourceCode;
    private String testCodeHash;

    public AssignmentTestRun(Assignment assignment, List<TestResult> results, String sourceCode, String testCodeHash) {
        this.assignment = assignment;
        this.results = results;
        this.sourceCode = sourceCode;
        this.testCodeHash = testCodeHash;
    }

}
